pkg/agent/openflow/pipeline.go

```diff
@@ -593,37 +616,166 @@ func (c *client) connectionTrackFlows(category cookie.Category) []binding.Flow {
+				// This flow is used to maintain DNAT conntrack for Service traffic.
+				connectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(proto).
+					Action().CT(false, connectionTrackTable.GetNext(), ctZone).NAT().CTDone().
+					Cookie(c.cookieAllocator.Request(category).Raw()).
+					Done(),
+				connectionTrackCommitTable.BuildFlow(priorityLow).MatchProtocol(proto).
+					MatchCTStateTrk(true).
+					MatchCTMark(ServiceCTMark, nil).
+					MatchRegRange(int(serviceLearnReg), marksRegServiceSelected, serviceLearnRegRange).
+					Cookie(c.cookieAllocator.Request(category).Raw()).
+					Action().GotoTable(connectionTrackCommitTable.GetNext()).
+					Done(),
+			)
+
+			if c.enabledProxyFull {
+				serviceConnectionTrackTable := c.pipeline[serviceConntrackTable]
+				serviceConnectionTrackCommitTable := c.pipeline[serviceConntrackCommitTable]
+				flows = append(flows,
+					// This flow is used to match the Service traffic from Antrea gateway. The Service traffic from gateway
+					// should arrive at table serviceConntrackCommitTable, otherwise it will be matched by other flows in
+					// table connectionTrackCommit.
+					connectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(proto).
+						MatchCTMark(ServiceCTMark, nil).
+						MatchRegRange(int(marksReg), markTrafficFromGateway, binding.Range{0, 15}).
+						Action().GotoTable(serviceConntrackCommitTable).
+						Cookie(c.cookieAllocator.Request(category).Raw()).
+						Done(),
+					// This flow is used to maintain SNAT conntrack for Service traffic.
+					serviceConnectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(proto).
+						Action().CT(false, serviceConnectionTrackTable.GetNext(), serviceCtZone).NAT().CTDone().
+						Cookie(c.cookieAllocator.Request(category).Raw()).
+						Done(),
+					// This flow is used to match the first packet of NodePort/LoadBalancer whose Endpoint is on
+					// host network and externalTrafficPolicy is Cluster. This packet requires SNAT.
+					// As the packet is from Antrea gateway, and it will pass through Antrea gateway, so a virtual
+					// Service IP is used to perform SNAT for the packet, rather than Antrea gateway's IP.
+					serviceConnectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(proto).
+						MatchRegRange(int(PortCacheReg), config.HostGatewayOFPort, ofPortRegRange).
+						MatchRegRange(int(serviceSnatReg), marksRegServiceNeedSNAT, serviceSnatMarkRange).
+						Cookie(c.cookieAllocator.Request(category).Raw()).
+						MatchCTStateNew(true).
+						MatchCTStateTrk(true).
+						MatchCTStateDNAT(true).
+						Action().CT(true, serviceConnectionTrackCommitTable.GetNext(), serviceCtZone).
+						SNAT(&binding.IPRange{StartIP: serviceVirtualIP, EndIP: serviceVirtualIP}, nil).
+						CTDone().
+						Done(),
+					// This flow is used to match the first packet of ClusterIP whose Endpoint is on host network.
+					// Since its source IP is Antrea gateway's IP, and it will pass through Antrea gateway.
```

Comment:
. This -> , this

Reply:
Reminder.

Reply:
Sorry, I may not get what you mean exactly. Is that you mean modify comment `// This flow is used to match the first packet of ClusterIP whose Endpoint is on host network.`?

Reply:
Got it, thanks!